app/Dockerfile

```diff
@@ -1,18 +1,39 @@
 FROM ubuntu:14.04
 
+# Some ENV variables
 ENV PATH="/mattermost/bin:${PATH}"
-
-RUN apt-get update && apt-get -y install curl netcat
-RUN mkdir -p /mattermost/data
-
 ENV MM_VERSION=3.9.0
 
-RUN curl https://releases.mattermost.com/$MM_VERSION/mattermost-team-$MM_VERSION-linux-amd64.tar.gz | tar -xvz
-
-RUN rm /mattermost/config/config.json
-COPY config.template.json /
-
+# Override default config
+ENV MM_SERVICESETTINGS_LISTENADDRESS=":80" \
+    MM_LOGSETTINGS_ENABLECONSOLE=false \
+    MM_LOGSETTINGS_CONSOLELEVEL="INFO" \
+    MM_FILESETTINGS_DIRECTORY="/mattermost/data/" \
+    MM_FILESETTINGS_ENABLEPUBLICLINK=true \
+    MM_FILESETTINGS_PUBLICLINKSALT="A705AklYF8MFDOfcwh3I488G8vtLlVip" \
+    MM_EMAILSETTINGS_SENDEMAILNOTIFICATIONS=false \
+    MM_EMAILSETTINGS_FEEDBACKEMAIL="" \
+    MM_EMAILSETTINGS_SMTPSERVER="" \
+    MM_EMAILSETTINGS_SMTPPORT="" \
+    MM_EMAILSETTINGS_INVITESALT="bjlSR4QqkXFBr7TP4oDzlfZmcNuH9YoS" \
+    MM_RATELIMITSETTINGS_ENABLE=true \
+    MM_SQLSETTINGS_DRIVERNAME="postgres"
+
+# Install some needed packages
+RUN apt-get update; \
```

Comment:
I think the best practice for joining the command is using "&&", which will cause the whole command fail if one of the failed. Otherwise, I think if the last command successfully completed, the whole command is considered successful.